controllers/front/ExportDataToJson.php

```
@@ -0,0 +1,175 @@
+                'date_add' => $cart['date_add'],
+                'products' => [],
+            ];
+
+            $products = $cart['products'];
+
+            foreach ($products as $product) {
+                $line['products'][] = [
+                    'reference' => $product['product_reference'],
+                    'name' => $product['product_name'],
+                    'quantity' => (int)$product['product_quantity'],
+                ];
+            }
+
+            $json['carts'][] = $line;
+        }
+
+        // MESSSAGES
+        $json['messages'] = [];
+
+        foreach ($messages as $message) {
+            $json['messages'][] = [
+                'ip' => $message['ip'],
+                'message' => $message['message'],
+                'date' => $message['date_add'],
+            ];
+        }
+
+        // CONNECTIONS
+        $json['connections'] = [];
+
+        foreach ($connections as $connection) {
+            $json['connections'][] = [
+                'http_referer' => $connection['http_referer'],
+                'page_viewed' => $connection['pages'],
+                'time_on_the_page' => $connection['time'],
+                'ip_address' => $connection['ipaddress'],
+                'date' => $connection['date_add'],
+            ];
+        }
+
+        // MODULES
+        $json['modules'] = $modules;
+
+        // Set the filename of the download
+        $filename = 'personalData-' . date('Y-m-d');
+
+        // Output CSV-specific headers
+        header('Content-Description: File Transfer');
+        header('Content-Type: application/json;');
+        header('Content-Disposition: attachment; filename="' . $filename . '.json";');
+        header('Content-Transfer-Encoding: binary');
+        header('Expires: 0');
+        header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+        header('Pragma: public');
+
+        $content = json_encode($json, JSON_PRETTY_PRINT);
+
+        exit($content);
```

a renderJson function already exist. exit is really bad :(

---

Whitch `renderJson` function?

---

oops I mean `ajaxRender` :)
If you add an exit to hide an error, it's not a good idea :/